### ISSUE TITLE
"Build Analysis" report is missing a build break in a test

### ISSUE THREAD

**Issue by markples** [dotnet/runtime#84049]
or perhaps the build break is being classified along with different known failures?  Note that the build break is due to an analyzer.

I don't know if these links will persist so here are links + screenshots.  This is the analysis: https://github.com/dotnet/runtime/pull/83895/checks?check_run_id=12264638203

![image](https://user-images.githubusercontent.com/25672677/228382238-a0506c52-05c1-4632-b683-042695d75948.png)

And this is a failure:
https://github.com/dotnet/runtime/pull/83895/checks?check_run_id=12262235297

![image](https://user-images.githubusercontent.com/25672677/228382361-8f431f95-4ecf-44be-b3a7-bf40ff47b407.png)

**Comment by hoyosjs** [comment]
@missymessa @AlitzelMendez is there any chance you can check this? The second error doesn't really match the blob on https://github.com/dotnet/runtime/issues/83655, but it's getting matched.

**Comment by hoyosjs** [comment]
@markples I fixed the issues. Now they actually surface.

**Comment by hoyosjs** [comment]
It doesn't explain the second issue matching, hence asking help from Ali

**Comment by AlitzelMendez** [comment]
At the moment that this build was analyzed, it matched the errors correctly. 

The problem was on the regex for the issue: https://github.com/dotnet/runtime/issues/83655

When this was analyzed the regex was:
`"ErrorPattern": "['WasmTestOnBrowser.+ END OF WORK ITEM LOG: Command timed out, and was killed]"` which was basically matching everything. 

This was fixed two days ago:
<img width="581" alt="image" src="https://user-images.githubusercontent.com/6895254/228659312-8cb13c22-6ce6-4de4-95ba-2e24e3bf356d.png">
but at that point this error probably matched a ton of builds, in fact if you see at the numbers from the last 7 days the number is unusually high (546 builds). 

When the error is updated, we analyzed the builds from the last 24 hours, but this build wasn't part of that timeframe so we didn't reanalyze the build.
